src/utils/promise-to-iterable.ts

```diff
@@ -0,0 +1,32 @@
+import { $$asyncIterator } from 'iterall';
+
+export function createIterableFromPromise<T>(promise: Promise<T>): AsyncIterator<T> {
+  let isResolved = false;
+
+  return {
+    next() {
+      if (isResolved) {
+        return this.return();
+      }
+
+      return promise
+        .then(value => ({ value, done: false }))
+        .catch(error => this.throw(error))
```

Comment:
This seems pretty strange - are you sure this is doing what you expect? The `return()` and `throw()` API on AsyncIterator are intended to be called by the consumer, not internally.

Since an AsyncIterator yields Promises, I would expect that it would simply return the wrapped promise once.

I would expect:

```js
import { $$asyncIterator } from 'iterall';

export function createIterableFromPromise<T>(promise: Promise<T>): AsyncIterator<T> {
  let isComplete = false;

  return {
    next() {
      if (!isComplete) {
        isComplete = true;
        return promise.then(value => { value, done: false });
      }
      return Promise.resolve({ value: undefined, done: true });
    },
    return() {
      isComplete = true;
      return Promise.resolve({ value: undefined, done: true });
    },
    throw(e: Error) {
      isComplete = true;
      throw e;
    },
    [$$asyncIterator]() {
      return this;
    },
  };
}
```